README


This archive contains the distribution Games-Risk-ExtraMaps-Imaginary,
version 3.112710:

  a set of imaginary maps for Games::Risk

This software is Copyright (c) 2011 by Jerome Quelin.

This is free software, licensed under:

  The GNU General Public License, Version 3, June 2007




Changes
Revision history for Games::Risk::ExtraMaps::Imaginary.

{{$NEXT}}

3.113460  2011-12-12 14:25:05 Europe/Paris
 - new map: lord of the ring
 - new map: ameroki

3.112710  2011-09-28 14:09:16 Europe/Paris
 - fix risk 2210 continent information

3.112691  2011-09-26 16:08:13 Europe/Paris
 - bumping major version to makesure cpan indexing works

1.112690  2011-09-26 15:34:47 Europe/Paris
 - initial release
 - maps included: godstorm, risk2210


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Module::Build" => "0.3601",
    "Test::More" => "0.88"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "a set of imaginary maps for Games::Risk",
  "dist_author" => [
    "Jerome Quelin"
  ],
  "dist_name" => "Games-Risk-ExtraMaps-Imaginary",
  "dist_version" => "3.112710",
  "license" => "gpl",
  "module_name" => "Games::Risk::ExtraMaps::Imaginary",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Games::Risk::ExtraMaps" => 0,
    "Games::Risk::Map" => 0,
    "Locale::Messages" => 0,
    "Locale::TextDomain" => 0,
    "Moose" => 0,
    "perl" => "5.010",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "share_dir" => {
    "dist" => "share"
  }
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Games/Risk/Map/Lotr.pm
use 5.010;
use strict;
use warnings;

package Games::Risk::Map::Lotr;
# ABSTRACT: Lord of the Rings Map

use Moose;
extends 'Games::Risk::Map';


use Locale::Messages   qw{ :locale_h bind_textdomain_filter turn_utf_8_on };
use Locale::TextDomain "Games-Risk-Map-Lotr";
use Moose;

extends 'Games::Risk::ExtraMaps::Imaginary';

my $domain ="Games-Risk-Map-Lotr";
bindtextdomain $domain, __PACKAGE__->localedir->stringify;
bind_textdomain_codeset $domain, "utf-8";
bind_textdomain_filter  $domain, sub { turn_utf_8_on($_[0]) };


# -- map  builders

sub name   { "lotr" }
sub title  { __("Lord of the Rings") }
sub author { "???" } # FIXME


# -- raw map information

sub _raw_continents {
return (
# id, name, bonus, color
#   0, __('Europe'), 5, blue
[1, __("Gondor"), 7, "#663300"],
[2, __("Mirkwood"), 4, "#009900"],
[3, __("Rohan"), 4, "#009999"],
[4, __("Rhovanion"), 5, "#ff9900"],
[5, __("Arnor"), 7, "#ff3333"],
[6, __("Eriador"), 3, "#cccc00"],
[7, __("Rhun"), 2, "#330033"],
[8, __("Haradwaith"), 2, "#ffff33"],
[9, __("Mordor"), 2, "#333300"],
);
}

sub _raw_countries {
return (
# greyscale, name, continent id, x, y, [connections]
#   1, __('Alaska'), 1, 43, 67, [ 1,2,3,38 ]
[1, __("Forodwaith"), 5, 13, 150, [49, 50, 2, 4, 3, 44, 46]],
[2, __("Eastern Angmar"), 5, 56, 154, [12, 1]],
[3, __("Borderlands"), 5, 78, 251, [1, 4, 43, 48, 5, 6, 10]],
[4, __("Angmar"), 5, 79, 210, [1, 3]],
[5, __("North Downs"), 5, 117, 251, [3, 6]],
[6, __("Fornost"), 5, 150, 258, [3, 10, 5, 8, 7]],
[7, __("Buckland"), 5, 188, 275, [6, 47, 8, 9]],
[8, __("Old Forest"), 5, 187, 249, [10, 6, 7, 9]],
[9, __("South Downs"), 5, 222, 249, [18, 10, 7, 8]],
[10, __("Weather Hills"), 5, 161, 225, [3, 11, 9, 8, 6]],
[11, __("Rhudaur"), 5, 148, 180, [17, 12, 10]],
[12, __("Carrock"), 2, 115, 140, [15, 11, 2, 13]],
[13, __("North Mirkwood"), 2, 112, 105, [14, 12, 51]],
[14, __("Eastern Mirkwood"), 2, 194, 90, [25, 16, 15, 13]],
[15, __("Anduin Valley"), 2, 227, 116, [26, 29, 16, 14, 12]],
[16, __("South Mirkwood"), 2, 260, 90, [25, 26, 15, 14]],
[17, __("Eregion"), 3, 220, 204, [19, 30, 11]],
[18, __("Minhiriath"), 3, 276, 295, [20, 19, 9, 44, 46, 37, 54]],
[19, __("Dunland"), 3, 282, 235, [20, 18, 17]],
[20, __("Enedwaith"), 3, 352, 251, [23, 19, 18]],
[21, __("West Rohan"), 3, 413, 266, [32, 23]],
[22, __("Fangorn"), 3, 356, 186, [23, 28, 27]],
[23, __("Gap of Rohan"), 3, 416, 190, [39, 21, 22, 27, 20]],
[24, __("Rhun Hills"), 4, 334, 15, [25]],
[25, __("Brown Lands"), 4, 354, 59, [31, 26, 24, 16, 14, 52]],
[26, __("Emyn Muil"), 4, 330, 102, [31, 27, 25, 16, 15]],
[27, __("The Wold"), 4, 347, 141, [23, 22, 28, 26]],
[28, __("Lorien"), 4, 287, 164, [22, 27, 29]],
[29, __("Gladden Fields"), 4, 220, 147, [30, 28, 15]],
[30, __("Moria"), 4, 232, 178, [17, 29]],
[31, __("Dead Marshes"), 4, 422, 111, [40, 59, 26, 25]],
[32, __("Druwaith Iaur"), 1, 478, 307, [34, 21]],
[33, __("Andras"), 1, 530, 322, [34]],
[34, __("Anfalas"), 1, 521, 284, [35, 32, 33]],
[35, __("Vale of Erech"), 1, 488, 252, [36, 34]],
[36, __("Lamedon"), 1, 497, 218, [38, 37, 35]],
[37, __("Belfalas"), 1, 548, 213, [38, 54, 36, 18]],
[38, __("Lebennin"), 1, 521, 177, [39, 37, 36]],
[39, __("Minis Tirith"), 1, 485, 167, [40, 23, 38]],
[40, __("Ithilien"), 1, 496, 133, [41, 39, 31, 62]],
[41, __("South Ithilien"), 1, 557, 134, [53, 40]],
[42, __("Forlindon"), 6, 120, 392, [44]],
[43, __("Lune Valley"), 6, 84, 316, [3, 44, 48]],
[44, __("Mithlond"), 6, 148, 359, [46, 42, 43, 1, 18]],
[45, __("Harlindon"), 6, 236, 368, [46]],
[46, __("Tower Hills"), 6, 162, 318, [47, 44, 45, 48, 1, 18]],
[47, __("The Shire"), 6, 202, 302, [7, 46]],
[48, __("Evendim Hills"), 6, 100, 291, [3, 43, 46]],
[49, __("North Rhun"), 7, 22, 16, [52, 50, 1]],
[50, __("Withered Heath"), 7, 54, 57, [49, 1, 51]],
[51, __("Esgaroth"), 7, 130, 63, [13, 50]],
[52, __("South Rhun"), 7, 173, 15, [25, 49]],
[53, __("Harandor"), 8, 589, 146, [56, 41]],
[54, __("Umbar"), 8, 627, 182, [55, 37, 18]],
[55, __("Deep Harad"), 8, 663, 173, [56, 54]],
[56, __("Harad"), 8, 646, 114, [57, 53, 55]],
[57, __("Near Harad"), 8, 644, 63, [58, 56]],
[58, __("Khand"), 8, 614, 17, [57]],
[59, __("Udun Value"), 9, 467, 88, [60, 31]],
[60, __("Mount Doom"), 9, 493, 68, [63, 61, 59]],
[61, __("Barad Dur"), 9, 478, 32, [63, 60]],
[62, __("Minas Morgul"), 9, 522, 95, [63, 40]],
[63, __("Gorgoroth"), 9, 521, 19, [64, 62, 61, 60]],
[64, __("Nurn"), 9, 564, 73, [63]],
);
}


sub _raw_cards {
return (
# type, id_country
#   artillery, 2
#   wildcard
["infantry", 1],
["artillery", 2],
["infantry", 3],
["infantry", 4],
["cavalry", 5],
["cavalry", 6],
["cavalry", 7],
["artillery", 8],
["artillery", 9],
["artillery", 10],
["cavalry", 11],
["artillery", 12],
["infantry", 13],
["infantry", 14],
["artillery", 15],
["artillery", 16],
["cavalry", 17],
["cavalry", 18],
["infantry", 19],
["cavalry", 20],
["infantry", 21],
["infantry", 22],
["cavalry", 23],
["artillery", 24],
["artillery", 25],
["infantry", 26],
["artillery", 27],
["cavalry", 28],
["cavalry", 29],
["infantry", 30],
["artillery", 31],
["infantry", 32],
["artillery", 33],
["cavalry", 34],
["infantry", 35],
["artillery", 36],
["infantry", 37],
["cavalry", 38],
["cavalry", 39],
["cavalry", 40],
["artillery", 41],
["infantry", 42],
["infantry", 43],
["artillery", 44],
["cavalry", 45],
["infantry", 46],
["artillery", 47],
["cavalry", 48],
["infantry", 49],
["artillery", 50],
["cavalry", 51],
["infantry", 52],
["artillery", 53],
["cavalry", 54],
["infantry", 55],
["artillery", 56],
["cavalry", 57],
["infantry", 58],
["artillery", 59],
["cavalry", 60],
["infantry", 61],
["artillery", 62],
["cavalry", 63],
["infantry", 64],
["joker", undef],
["joker", undef],
);
}

sub _raw_missions {
return (
# id player to destroy, nb coutnry to occupy + min armies, 3 x id of continents to occupy, description
#   0, 0,0,5,2,0,__("Conquer the continents of ASIA and SOUTH AMERICA.")
#   0, 0,0,3,6,*,__("Conquer the continents of EUROPE and AUSTRALIA and a third continent of your choice.")
#   0,18,2,0,0,0,__("Occupy 18 countries of your choice and occupy each with at least 2 armies.")
#   0,24,1,0,0,0,__("Occupy 24 countries of your choice and occupy each with at least 1 army.")
#   1,24,1,0,0,0,__("Destroy all of PLAYER1's TROOPS. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries.")

);
}

no Moose;
__PACKAGE__->meta->make_immutable;

1;
__END__

=head1 DESCRIPTION

LOTR Map by ???


lib/Games/Risk/Map/Ameroki.pm
use 5.010;
use strict;
use warnings;

package Games::Risk::Map::Ameroki;
# ABSTRACT: Ameroki Map

use Moose;
extends 'Games::Risk::Map';


use Locale::Messages   qw{ :locale_h bind_textdomain_filter turn_utf_8_on };
use Locale::TextDomain "Games-Risk-Map-Ameroki";
use Moose;

extends 'Games::Risk::ExtraMaps::Imaginary';

my $domain ="Games-Risk-Map-Ameroki";
bindtextdomain $domain, __PACKAGE__->localedir->stringify;
bind_textdomain_codeset $domain, "utf-8";
bind_textdomain_filter  $domain, sub { turn_utf_8_on($_[0]) };


# -- map  builders

sub name   { "ameroki" }
sub title  { __("Ameroki Map") }
sub author { "map: ameroki.map" }


# -- raw map information

sub _raw_continents {
return (
# id, name, bonus, color
#   0, __('Europe'), 5, blue
[1, "azio", 5, "#9aff80"],
[2, "ameroki", 10, "yellow"],
[3, "utropa", 10, "#a980ff"],
[4, "amerpoll", 5, "red"],
[5, "afrori", 5, "#ffd780"],
[6, "ulstrailia", 5, "magenta"],
);
}

sub _raw_countries {
return (
# greyscale, name, continent id, x, y, [connections]
#   1, __('Alaska'), 1, 43, 67, [ 1,2,3,38 ]
[1, "siberia", 1, 329, 152, [20, 32, 2]],
[2, "worrick", 1, 308, 199, [1, 10, 35, 3]],
[3, "yazteck", 1, 284, 260, [2, 4, 6, 35]],
[4, "kongrolo", 1, 278, 295, [3, 35, 5, 6]],
[5, "china", 1, 311, 350, [4, 6, 42, 7, 9]],
[6, "middle east", 1, 339, 299, [3, 5, 9, 4]],
[7, "sluci", 1, 358, 376, [5, 9, 8]],
[8, "afganistan", 1, 400, 350, [9, 7]],
[9, "kancheria", 1, 388, 313, [5, 8, 7, 6, 10]],
[10, "india", 1, 405, 229, [11, 2, 9]],
[11, "japan", 1, 462, 236, [17, 10]],
[12, "new guinia", 6, 513, 314, [15, 13]],
[13, "western ulstrailia", 6, 525, 340, [15, 14, 12]],
[14, "eastern ulstarilia", 6, 567, 355, [15, 13]],
[15, "jacuncail", 6, 561, 300, [17, 12, 13, 14]],
[16, "tungu", 5, 537, 178, [18, 17, 19]],
[17, "south afrori", 5, 538, 210, [16, 15, 11]],
[18, "north afrori", 5, 486, 131, [20, 19, 16]],
[19, "east afrori", 5, 540, 144, [18, 21, 16]],
[20, "egypt", 5, 443, 100, [23, 1, 18]],
[21, "maganar", 5, 578, 110, [19, 27]],
[22, "pero", 4, 374, 25, [24, 23]],
[23, "heaurt", 4, 428, 32, [22, 24, 20]],
[24, "vagnagale", 4, 327, 40, [25, 22, 23]],
[25, "argentina", 4, 277, 36, [30, 24]],
[26, "ireland", 3, 128, 127, [27, 31]],
[27, "ihesia", 3, 130, 96, [26, 28, 21, 31]],
[28, "western utropa", 3, 181, 112, [27, 31, 29, 32, 30]],
[29, "souther utropa", 3, 235, 133, [28, 32, 30]],
[30, "northern utropa", 3, 253, 82, [28, 25, 29]],
[31, "senadlavin", 3, 177, 155, [28, 34, 32, 26, 27]],
[32, "great britain", 3, 244, 168, [28, 1, 34, 29, 31]],
[33, "teramar", 2, 103, 249, [34, 36]],
[  34,  "western united states",  2,  164,  235,  [38, 35, 33, 36, 37, 31, 32],],
[35, "czeck", 2, 222, 257, [2, 4, 38, 34, 3]],
[36, "alberta", 2, 122, 295, [34, 37, 41, 33, 39]],
[37, "central ameroki", 2, 182, 329, [40, 38, 34, 36, 39]],
[38, "albania", 2, 205, 305, [40, 35, 34, 37]],
[39, "duiestie", 2, 164, 355, [40, 37, 41, 36]],
[40, "Quebeck", 2, 203, 366, [42, 39, 37, 38]],
[41, "vinenlant", 2, 128, 373, [36, 39]],
[42, "heal", 1, 271, 397, [5, 40]],
);
}


sub _raw_cards {
return (
# type, id_country
#   artillery, 2
#   wildcard
[__("cavalry"), 1],
[__("infantry"), 2],
[__("artillery"), 3],
[__("cavalry"), 4],
[__("infantry"), 5],
[__("artillery"), 6],
[__("cavalry"), 7],
[__("infantry"), 8],
[__("artillery"), 9],
[__("cavalry"), 10],
[__("infantry"), 11],
[__("artillery"), 12],
[__("cavalry"), 13],
[__("infantry"), 14],
[__("artillery"), 15],
[__("cavalry"), 16],
[__("infantry"), 17],
[__("artillery"), 18],
[__("cavalry"), 19],
[__("infantry"), 20],
[__("artillery"), 21],
[__("cavalry"), 22],
[__("infantry"), 23],
[__("artillery"), 24],
[__("cavalry"), 25],
[__("infantry"), 26],
[__("artillery"), 27],
[__("cavalry"), 28],
[__("infantry"), 29],
[__("artillery"), 30],
[__("cavalry"), 31],
[__("infantry"), 32],
[__("artillery"), 33],
[__("cavalry"), 34],
[__("artillery"), 35],
[__("cavalry"), 36],
[__("infantry"), 37],
[__("artillery"), 38],
[__("artillery"), 39],
[__("cavalry"), 40],
[__("infantry"), 41],
[__("artillery"), 42],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
[__("joker"), undef],
);
}

sub _raw_missions {
return (
# id player to destroy, nb coutnry to occupy + min armies, 3 x id of continents to occupy, description
#   0, 0,0,5,2,0,__("Conquer the continents of ASIA and SOUTH AMERICA.")
#   0, 0,0,3,6,*,__("Conquer the continents of EUROPE and AUSTRALIA and a third continent of your choice.")
#   0,18,2,0,0,0,__("Occupy 18 countries of your choice and occupy each with at least 2 armies.")
#   0,24,1,0,0,0,__("Occupy 24 countries of your choice and occupy each with at least 1 army.")
#   1,24,1,0,0,0,__("Destroy all of PLAYER1's TROOPS. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries.")

);
}

no Moose;
__PACKAGE__->meta->make_immutable;

1;
__END__

=head1 DESCRIPTION

Ameroki Map by map: ameroki.map.


lib/Games/Risk/ExtraMaps/Imaginary.pm
#
# This file is part of Games-Risk-ExtraMaps-Imaginary
#
# This software is Copyright (c) 2011 by Jerome Quelin.
#
# This is free software, licensed under:
#
#   The GNU General Public License, Version 3, June 2007
#
use 5.010;
use strict;
use warnings;

package Games::Risk::ExtraMaps::Imaginary;
{
  $Games::Risk::ExtraMaps::Imaginary::VERSION = '3.112710';
}
# ABSTRACT: a set of imaginary maps for Games::Risk

use Moose;
extends 'Games::Risk::ExtraMaps';

sub extra_category { "Imaginary" }

__PACKAGE__->meta->make_immutable;
1;


=pod

=head1 NAME

Games::Risk::ExtraMaps::Imaginary - a set of imaginary maps for Games::Risk

=head1 VERSION

version 3.112710

=head1 DESCRIPTION

This distribution holds a few imaginary maps.

=head1 AUTHOR

Jerome Quelin

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2011 by Jerome Quelin.

This is free software, licensed under:

  The GNU General Public License, Version 3, June 2007

=cut


__END__



lib/Games/Risk/Map/Risk2210.pm
#
# This file is part of Games-Risk-ExtraMaps-Imaginary
#
# This software is Copyright (c) 2011 by Jerome Quelin.
#
# This is free software, licensed under:
#
#   The GNU General Public License, Version 3, June 2007
#
use 5.010;
use strict;
use warnings;

package Games::Risk::Map::Risk2210;
{
  $Games::Risk::Map::Risk2210::VERSION = '3.112710';
}
# ABSTRACT: Risk 2210 Map

use Moose;
extends 'Games::Risk::Map';


use Locale::Messages   qw{ :locale_h bind_textdomain_filter turn_utf_8_on };
use Locale::TextDomain "Games-Risk-Map-Risk2210";
use Moose;

extends 'Games::Risk::ExtraMaps::Imaginary';

my $domain ="Games-Risk-Map-Risk2210";
bindtextdomain $domain, __PACKAGE__->localedir->stringify;
bind_textdomain_codeset $domain, "utf-8";
bind_textdomain_filter  $domain, sub { turn_utf_8_on($_[0]) };


# -- map  builders

sub name   { "risk2210" }
sub title  { __("Risk 2210 Map") }
sub author { "Matthias Kuehl" }


# -- raw map information

sub _raw_continents {
return (
# id, name, bonus, color
#   0, __('Europe'), 5, blue
[1, __("North America"), 5, "yellow"],
[2, __("South America"), 2, "red"],
[3, __("Europe"), 5, "blue"],
[4, __("Africa"), 3, "orange"],
[5, __("Asia"), 7, "green"],
[6, __("Australia"), 2, "magenta"],
[7, __("Asia Pacific"), 1, "yellow"],
[8, __("US Pacific"), 2, "blue"],
[9, __("North Atlantic"), 2, "red"],
[10, __("South Atlantic"), 1, "green"],
[11, __("Indian"), 2, "orange"],
);
}

sub _raw_countries {
return (
# greyscale, name, continent id, x, y, [connections]
#   1, __('Alaska'), 1, 43, 67, [ 1,2,3,38 ]
[1, __("Northwestern Oil Emirate"), 1, 47, 19, [2, 4, 38]],
[2, __("Nunavut"), 1, 105, 23, [1, 4, 5, 3]],
[3, __("Exiled States of America"), 1, 222, 23, [2, 6, 14]],
[4, __("Alberta"), 1, 85, 57, [1, 2, 5, 7]],
[5, __("Canada"), 1, 125, 67, [2, 4, 6, 7, 8]],
[6, __("Republique du Quebec"), 1, 175, 75, [3, 5, 8]],
[7, __("Continental Biospheres"), 1, 80, 100, [4, 5, 8, 9, 45]],
[8, __("American Republic"), 1, 117, 121, [5, 6, 7, 9, 49]],
[9, __("Mexico"), 1, 81, 161, [7, 8, 10]],
[10, __("Nuevo Timoto"), 2, 115, 216, [9, 11, 12, 47]],
[11, __("Andean Nations"), 2, 94, 257, [10, 12, 13]],
[12, __("Amazon Desert"), 2, 145, 265, [10, 11, 13, 21, 50, 51]],
[13, __("Argentina"), 2, 90, 325, [11, 12]],
[14, __("Iceland GRC"), 3, 268, 47, [3, 17, 15]],
[15, __("Jotenheim"), 3, 328, 28, [16, 18, 14, 17]],
[16, __("Ukrayina"), 3, 412, 70, [15, 28, 30, 31, 18, 20]],
[17, __("New Avalon"), 3, 272, 84, [14, 18, 19, 15, 48]],
[18, __("Warsaw Republic"), 3, 333, 89, [15, 16, 17, 19, 20]],
[19, __("Andorra"), 3, 270, 132, [17, 18, 20, 21]],
[  20,  __("Imperial Balkania"),  3,  340,  116,  [16, 18, 19, 21, 22, 31],],
[  21,  __("Saharan Empire"),  4,  280,  211,  [12, 19, 20, 22, 23, 24, 52],],
[22, __("Egypt"), 4, 350, 180, [20, 21, 24, 31]],
[23, __("Zaire Military Zone"), 4, 349, 274, [21, 24, 25]],
[24, __("Ministry of Djibouti"), 4, 395, 245, [21, 22, 23, 25, 26]],
[25, __("Lesotho"), 4, 357, 347, [23, 24, 26]],
[26, __("Madagascar"), 4, 417, 332, [24, 25, 54]],
[27, __("Siberia"), 5, 520, 32, [28, 29, 34, 35, 36]],
[28, __("Enclave of the Bear"), 5, 475, 65, [16, 27, 29, 30]],
[29, __("Hong Kong"), 5, 555, 145, [27, 28, 30, 32, 33, 36, 44]],
[30, __("Afghanistan"), 5, 466, 117, [16, 28, 29, 31, 32]],
[31, __("Middle East"), 5, 427, 179, [16, 22, 30, 32, 20]],
[32, __("United Indiastan"), 5, 511, 189, [29, 30, 31, 33, 53]],
[33, __("Angkhor Wat"), 5, 577, 207, [29, 32, 39]],
[34, __("Sakha"), 5, 580, 23, [27, 35, 38]],
[35, __("Alden"), 5, 570, 70, [27, 34, 36, 38]],
[  36,  __("Khan Industrial State"),  5,  577,  107,  [29, 35, 37, 38, 27],],
[37, __("Japan"), 5, 650, 114, [36, 38, 44]],
[38, __("Pevek"), 5, 630, 26, [1, 34 .. 37]],
[39, __("Java Cartel"), 6, 605, 270, [33, 41, 40, 43]],
[40, __("New Guinea"), 6, 660, 284, [39, 41, 42]],
[41, __("Aboriginal League"), 6, 595, 365, [39, 40, 42, 55]],
[42, __("Australian Testing Ground"), 6, 650, 352, [40, 41]],
[43, __("Sung Tzu"), 7, 652, 228, [39, 44]],
[44, __("Neo Tokyo"), 7, 648, 172, [29, 37, 43, 47]],
[45, __("Poseidon"), 8, 31, 88, [7, 46]],
[46, __("Hawaiian Preserve"), 8, 33, 145, [45, 47]],
[47, __("New Atlantis"), 8, 49, 198, [10, 44, 46]],
[48, __("Western Ireland"), 9, 217, 133, [17, 49]],
[49, __("New York"), 9, 163, 159, [8, 48, 50]],
[50, __("Nova Brasilia"), 9, 187, 212, [12, 49]],
[51, __("Neo Paulo"), 10, 206, 307, [12, 52]],
[52, __("The Ivory Reef"), 10, 269, 293, [21, 51]],
[53, __("South Ceylon"), 11, 483, 268, [32, 54]],
[54, __("Microcorp"), 11, 473, 327, [26, 53, 55]],
[55, __("Akara"), 11, 527, 359, [41, 54]],
);
}


sub _raw_cards {
return (
# type, id_country
#   artillery, 2
#   wildcard
["infantry", 1],
["artillery", 2],
["infantry", 3],
["infantry", 4],
["cavalry", 5],
["cavalry", 6],
["cavalry", 7],
["artillery", 8],
["artillery", 9],
["artillery", 10],
["cavalry", 11],
["artillery", 12],
["infantry", 13],
["infantry", 14],
["artillery", 15],
["artillery", 16],
["cavalry", 17],
["cavalry", 18],
["infantry", 19],
["cavalry", 20],
["infantry", 21],
["infantry", 22],
["cavalry", 23],
["artillery", 24],
["artillery", 25],
["infantry", 26],
["artillery", 27],
["cavalry", 28],
["cavalry", 29],
["infantry", 30],
["artillery", 31],
["infantry", 32],
["artillery", 33],
["cavalry", 34],
["infantry", 35],
["artillery", 36],
["infantry", 37],
["cavalry", 38],
["cavalry", 39],
["cavalry", 40],
["artillery", 41],
["infantry", 42],
["artillery", 43],
["artillery", 44],
["infantry", 45],
["artillery", 46],
["cavalry", 47],
["cavalry", 48],
["infantry", 49],
["artillery", 50],
["infantry", 51],
["artillery", 52],
["cavalry", 53],
["infantry", 54],
["infantry", 55],
["joker", undef],
["joker", undef],
);
}

sub _raw_missions {
return (
# id player to destroy, nb coutnry to occupy + min armies, 3 x id of continents to occupy, description
#   0, 0,0,5,2,0,__("Conquer the continents of ASIA and SOUTH AMERICA.")
#   0, 0,0,3,6,*,__("Conquer the continents of EUROPE and AUSTRALIA and a third continent of your choice.")
#   0,18,2,0,0,0,__("Occupy 18 countries of your choice and occupy each with at least 2 armies.")
#   0,24,1,0,0,0,__("Occupy 24 countries of your choice and occupy each with at least 1 army.")
#   1,24,1,0,0,0,__("Destroy all of PLAYER1's TROOPS. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries.")
[  0,  0,  0,  5,  2,  0,  __("Conquer the continents of Asia and South America."),],
[  0,  0,  0,  3,  6,  "*",  __("Conquer the continents of Europe and Australia and a third continent of your choice."),],
[0, 0, 0, 5, 4, 0, __("Conquer the continents of Asia and Africa.")],
[  0,  0,  0,  3,  2,  "*",  __("Conquer the continents of Europe and South America and a third continent of your choice."),],
[  0,  0,  0,  1,  6,  0,  __("Conquer the continents of North America and Australia."),],
[  0,  0,  0,  1,  4,  0,  __("Conquer the continents of North America and Africa."),],
[  0,  18,  2,  0,  0,  0,  __("Occupy 18 countries of your choice and occupy each with at least 2 armies."),],
[  0,  24,  1,  0,  0,  0,  __("Occupy 24 countries of your choice and occupy each with at least 1 army."),],
[  1,  24,  1,  0,  0,  0,  __("Destroy all of Player1's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
[  2,  24,  1,  0,  0,  0,  __("Destroy all of Player2's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
[  3,  24,  1,  0,  0,  0,  __("Destroy all of Player3's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
[  4,  24,  1,  0,  0,  0,  __("Destroy all of Player4's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
[  5,  24,  1,  0,  0,  0,  __("Destroy all of Player5's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
[  6,  24,  1,  0,  0,  0,  __("Destroy all of Player6's Troops. If they are yours or they have already been destroyed by another player then your mission is: Occupy 24 countries."),],
);
}

no Moose;
__PACKAGE__->meta->make_immutable;

1;


=pod

=head1 NAME

Games::Risk::Map::Risk2210 - Risk 2210 Map

=head1 VERSION

version 3.112710

=head1 DESCRIPTION

Risk 2210 Map by Matthias Kuehl.

=head1 AUTHOR

Jerome Quelin

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2011 by Jerome Quelin.

This is free software, licensed under:

  The GNU General Public License, Version 3, June 2007

=cut


__END__

